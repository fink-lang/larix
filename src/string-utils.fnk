unindent_text = fn text, ind:
  [first, ...rest] = text.split rx/\n/g

  [...lines] = pipe rest:
    map line:
      line.slice ind

  [first, ...lines].join '\n'
