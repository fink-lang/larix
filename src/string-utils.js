import {curr_loc, next_loc} from '@fink/prattler';


export const unindent_text = (text, ind)=> (
  text
    .split(/\n/g)
    .map((line, idx)=> line.slice(idx > 0 ? ind : 0))
    .join('\n')
);


export const strip_block = (strings, ...parts)=> {
  const [, ...lines] = String.raw({raw: strings}, ...parts).split('\n');
  const ind = lines[0].search(/[^ ]/);
  const str = lines.map((lne)=> lne.slice(ind)).join('\n');
  return str;
};

