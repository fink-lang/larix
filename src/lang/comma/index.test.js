import {parse_expr} from '../..';
import {strip_block} from '../../string-utils';


describe('comma', ()=> {
  it('parses single line: fn foo, bar: foo', ()=> {
    expect(
      parse_expr(`foo, bar`)
    ).toMatchSnapshot();
  });
});
