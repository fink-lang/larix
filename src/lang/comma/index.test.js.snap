// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`comma parses single line: fn foo, bar: foo 1`] = `ident (1:0-1:3) foo`;
